browser/plugins/network-breadcrumbs.js

```diff
@@ -0,0 +1,144 @@
+/* global XMLHttpRequest */
+const breadcrumbType = 'network'
+let restoreFunctions = []
+let client
+
+/*
+ * Leaves breadcrumbs when network requests occur
+ */
+module.exports = {
+  init: (_client) => {
+    client = _client
+    monkeyPatchXMLHttpRequest()
+    monkeyPatchFetch()
+  },
+  destroy: () => {
+    restoreFunctions.forEach(restore => restore())
+    restoreFunctions = []
+  },
+  configSchema: {
+    networkBreadcrumbsEnabled: {
+      defaultValue: () => undefined,
+      validate: (value) => value === true || value === false || value === undefined,
+      message: '(boolean) networkBreadcrumbsEnabled should be true or false'
+    }
+  }
+}
+
+// ------------------------------------------------------------------------------------------------
+// XMLHttpRequest and XDomainRequest monkey patch
+// ------------------------------------------------------------------------------------------------
+// create keys to safely store metadata on the request object
+const requestSetupKey = 'BUGSNAG:SETUP'
+const requestUrlKey = 'BUGSNAG:REQUEST_URL'
+const requestMethodKey = 'BUGSNAG:REQUEST_METHOD'
+
+// generic function for monkey patching both XMLHttpRequest and XDomainRequest
+function monkeyPatchXMLHttpRequest () {
+  if (!('addEventListener' in XMLHttpRequest.prototype)) return
+  let nativeOpen = XMLHttpRequest.prototype.open
+
+  // override native open()
+  XMLHttpRequest.prototype.open = function open (method, url) {
+    // store url and HTTP method for later
+    this[requestUrlKey] = url
+    this[requestMethodKey] = method
+
+    // if we have already setup listeners, it means open() was called twice, we need to remove
```

Comment:
how did you discover this edge case?

Reply:
I tested this behavior in the browser console 
![demo](https://user-images.githubusercontent.com/555044/38153429-998fec28-3421-11e8-96a0-5396ec933188.png)